cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -71,24 +90,18 @@ public void run() {
                                           public void handle(NamespaceId namespaceId, final Location location,
                                                              final String namespacedLogBaseDir) {
                                             try {
-                                              impersonator.doAs(namespaceId, new Callable<Void>() {
-                                                @Override
-                                                public Void call() throws Exception {
-                                                  if (location.exists()) {
-                                                    LOG.info("Deleting log file {}", location);
-                                                    location.delete();
-                                                    parentDirs.put(namespacedLogBaseDir, getParent(location));
-                                                  }
-                                                  return null;
-                                                }
-                                              });
+                                              deleteLogFiles(parentDirs, namespaceId, namespacedLogBaseDir, location);
                                               namespacedLogBaseDirMap.put(namespacedLogBaseDir, namespaceId);
                                             } catch (Exception e) {
                                               LOG.error("Got exception when deleting path {}", location, e);
                                               throw Throwables.propagate(e);
                                             }
                                           }
                                         });
+
+      // clean log files which does not have corresponding meta data
+      cleanFilesWithoutMeta(tillTime, namespacedLogBaseDirMap, parentDirs);
```

Comment:
It would be good to wrap this method in a try block and catch and log exceptions, so that any exceptions thrown here does not affect regular deletion.